Gems/ROS2/Code/Source/RobotImporter/URDF/JointsMaker.cpp

```diff
@@ -0,0 +1,143 @@
+
+        AZ_Printf(
+            "JointsMaker",
+            "Quaternion from URDF to o3de %f, %f, %f, %f",
+            quaternion.GetX(),
+            quaternion.GetY(),
+            quaternion.GetZ(),
+            quaternion.GetW());
+        const AZ::Vector3 rotation = quaternion.GetEulerDegrees();
+
+        switch (joint->type)
+        {
+        case urdf::Joint::FIXED:
+            {
+                jointComponent = followColliderEntity->CreateComponent<PhysX::EditorFixedJointComponent>();
+            }
+            break;
+        case urdf::Joint::PRISMATIC:
+            {
+                jointComponent = followColliderEntity->CreateComponent<PhysX::EditorPrismaticJointComponent>();
+                followColliderEntity->Activate();
+
+                PhysX::EditorJointRequestBus::Event(
+                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
+                    &PhysX::EditorJointRequests::SetVector3Value,
+                    PhysX::JointsComponentModeCommon::ParamaterNames::Rotation,
+                    rotation);
+
+                PhysX::EditorJointRequestBus::Event(
+                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
+                    &PhysX::EditorJointRequests::SetLinearValuePair,
+                    PhysX::JointsComponentModeCommon::ParamaterNames::LinearLimits,
+                    PhysX::AngleLimitsFloatPair(joint->limits->upper, joint->limits->lower));
+
+                followColliderEntity->Deactivate();
+            }
+            break;
+        case urdf::Joint::CONTINUOUS:
+            { // Implemented as Hinge with angular limit disabled
+                jointComponent = followColliderEntity->CreateComponent<PhysX::EditorHingeJointComponent>();
+                followColliderEntity->Activate();
+
+                PhysX::EditorJointRequestBus::Event(
+                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
+                    &PhysX::EditorJointRequests::SetVector3Value,
+                    PhysX::JointsComponentModeCommon::ParamaterNames::Rotation,
+                    rotation);
+
+                PhysX::EditorJointRequestBus::Event(
+                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
+                    &PhysX::EditorJointRequests::SetLinearValuePair,
+                    PhysX::JointsComponentModeCommon::ParamaterNames::TwistLimits,
+                    PhysX::AngleLimitsFloatPair(AZ::RadToDeg(AZ::Constants::TwoPi), -AZ::RadToDeg(AZ::Constants::TwoPi)));
+
+                PhysX::EditorJointRequestBus::Event(
+                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
+                    &PhysX::EditorJointRequests::SetBoolValue,
+                    PhysX::JointsComponentModeCommon::ParamaterNames::EnableLimits,
+                    false);
```

Comment:
pro-tip: EBus calls can accept a functor in addition to a function pointer.

The benefit of using a functor for this code, is needing to go through the EBus machinery 3 times can be avoided which improves performance.

This is especially useful if the EBus protects the event functions with a mutex. For the EditorJoinRequestBus that is not the case.

 ```suggestion
                PhysX::EditorJointRequestBus::Event(
                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
                    [&rotation](Phys::EditorJointRequests* editorJointRequest)
                    {
                        editorJointRequest->SetVector3Value(PhysX::JointsComponentModeCommon::ParamaterNames::Rotation, rotation);
                        editorJointRequest->SetLinearValuePair(PhysX::JointsComponentModeCommon::ParamaterNames::TwistLimits,
                            PhysX::AngleLimitsFloatPair(AZ::RadToDeg(AZ::Constants::TwoPi), -AZ::RadToDeg(AZ::Constants::TwoPi));
                        editorJointRequest->SetBoolValue(PhysX::JointsComponentModeCommon::ParamaterNames::EnableLimits, false);
                    },
                    PhysX::JointsComponentModeCommon::ParamaterNames::EnableLimits,
                    false);
```

Reply:
Amazing tip, done